Description: Link lldb with -latomic to get GCC atomic helpers
Author: Adam Conrad <adconrad@ubuntu.com>
Forwarded: no
Last-Update: 2013-08-21

--- llvm-toolchain-3.3-3.3.orig/lldb/lib/Makefile
+++ llvm-toolchain-3.3-3.3/lldb/lib/Makefile
@@ -140,6 +140,10 @@ ifeq ($(HOST_OS), $(filter $(HOST_OS), L
     LLVMLibsOptions += -Wl,--no-undefined
     # Link in python
     LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt
+    ifneq (,$(filter $(shell dpkg-architecture -qDEB_HOST_ARCH), mips mipsel powerpc powerpcspe))
+    # Link GCC atomic helper library
+    LLVMLibsOptions += -latomic
+    endif
     LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT)
 endif
 
